--- slrn-0.9.7.4/configure.in.orig	Thu Mar 21 22:54:15 2002
+++ slrn-0.9.7.4/configure.in	Thu Mar 21 22:55:36 2002
@@ -90,22 +90,22 @@
 dnl the setgid patch
 CF_COMPILE_OPTION(setgid-code,
     [  --enable-setgid-code    Allow running slrnpull as an unpriviledged user],
-    [ SLRNPULL_USE_SETGID_POSTS ])
+    [ SLRNPULL_USE_SETGID_POSTS])
 
 dnl NNTP support
 CF_COMPILE_OPTION(nntp,
     [  --disable-nntp          Disable NNTP support],
-    [ SLRN_HAS_NNTP_SUPPORT ])
+    [ SLRN_HAS_NNTP_SUPPORT])
 
 dnl news spool support
 CF_COMPILE_OPTION(spool,
     [  --enable-spool          Enable support for reading directly from spool],
-    [ SLRN_HAS_SPOOL_SUPPORT ])
+    [ SLRN_HAS_SPOOL_SUPPORT])
 
 dnl inews support
 CF_COMPILE_OPTION(inews,
     [  --enable-inews          Enable inews posting support],
-    [ SLRN_HAS_INEWS_SUPPORT ])
+    [ SLRN_HAS_INEWS_SUPPORT])
 
 if test "x$enable_inews" = xyes ; then
 
@@ -126,22 +126,22 @@
 dnl enforce the use of inews
 CF_COMPILE_OPTION(force-inews,
     [  --enable-force-inews    Enforce the use of inews],
-    [ SLRN_FORCE_INEWS ])
+    [ SLRN_FORCE_INEWS])
 
 dnl disallow custom From headers
 CF_COMPILE_OPTION(strict-from,
     [  --enable-strict-from    Disallow custom From headers],
-    [ SLRN_HAS_STRICT_FROM ])
+    [ SLRN_HAS_STRICT_FROM])
 
 dnl embed slang interpreter
 CF_COMPILE_OPTION(slang,
     [  --disable-slang         Don't make the slang interpreter available],
-    [ SLRN_HAS_SLANG ])
+    [ SLRN_HAS_SLANG])
 
 dnl disable MIME support
 CF_COMPILE_OPTION(mime,
     [  --disable-mime          Disable MIME support],
-    [ SLRN_HAS_MIME ])
+    [ SLRN_HAS_MIME])
 
 dnl character set mapping
 CF_COMPILE_OPTION(charmap,
@@ -173,12 +173,12 @@
     [\
   --enable-mid-cache      Make slrn cache Message-IDs to eliminate cross-posts
                           (only needed when server has no xref headers)],
-    [ SLRN_HAS_MSGID_CACHE ])
+    [ SLRN_HAS_MSGID_CACHE])
 
 dnl Use In-Reply-To if necessary
 CF_COMPILE_OPTION(fake-refs,
     [  --disable-fake-refs     Don't use In-Reply-To if References isn't available],
-    [ SLRN_HAS_FAKE_REFS ])
+    [ SLRN_HAS_FAKE_REFS])
 
 dnl enable grouplens support
 CF_COMPILE_OPTION(grouplens,
