diff -Nur slrn-0.9.6.4pl4c.orig/src/slrnconf.h slrn-0.9.6.4pl4c/src/slrnconf.h
--- slrn-0.9.6.4pl4c.orig/src/slrnconf.h	Fri Mar 16 21:38:43 2001
+++ slrn-0.9.6.4pl4c/src/slrnconf.h	Fri Mar 30 14:17:50 2001
@@ -49,13 +49,13 @@
 #define SLRN_HAS_VERBATIM_MARKS		1
 
 /* define if you want grouplens (R.I.P.) support */
-#define SLRN_HAS_GROUPLENS		0
+#define SLRN_HAS_GROUPLENS		1
 
 /* define if you want slrn to generate Message-IDs */
 #define SLRN_HAS_GEN_MSGID		1
 
 /* define to make slrn cache Message-IDs to eliminate cross-posts */
-#define SLRN_HAS_MSGID_CACHE		0
+#define SLRN_HAS_MSGID_CACHE		1
 
 /* define if slrn should use In-Reply-To if no References are available */
 #define SLRN_HAS_FAKE_REFS		1
diff -Nur slrn-0.9.7.3.orig/src/slrnfeat.h slrn-0.9.7.3/src/slrnfeat.h
--- slrn-0.9.7.3.orig/src/slrnfeat.h	Sun Mar 11 21:55:57 2001
+++ slrn-0.9.7.3/src/slrnfeat.h	Fri Mar 30 14:21:50 2001
@@ -59,8 +59,8 @@
   /* Note the -S flag.  slrn appends the signature to the file to be posted and
    * the -S flag tells inews not to also do this.  The -h flag must be used. */

   /* Changing this here has no effect when using ./configure (on Unix) */
-# define SLRN_INEWS_COMMAND      "/usr/local/bin/inews -S -h"
+# define SLRN_INEWS_COMMAND      "/usr/bin/inews -S -h"
 #endif
 
 #if SLRN_HAS_SPOOL_SUPPORT || defined(SLRNPULL_CODE)
@@ -162,7 +162,7 @@
 /* #define OUR_ORGANIZATION "organization-name" */
 /* #define OUR_HOSTNAME "host.name.here" */
 /* #define NNTPSERVER_NAME  "my.server.name" */
-/* #define NNTPSERVER_FILE		"/usr/local/lib/news/nntp_server" */
+#define NNTPSERVER_FILE		"/etc/nntpserver"
 
 #if SLRN_HAS_GROUPLENS
 # undef SLRN_USE_SLTCP
@@ -172,7 +172,7 @@
 /* The rest of the files apply to slrnpull. */
 
 /* This must be set to an absolute pathname. */
-#define SLRNPULL_ROOT_DIR	"/var/spool/news/slrnpull"
+#define SLRNPULL_ROOT_DIR	"/var/spool/slrnpull"
 
 /* The remaing variables are specified as relative names with respect to the
  * SLRNPULL_ROOT_DIR.
