--- slrn-0.9.8.1pl1/src/art.c.orig	2005-01-29 00:26:09.000000000 +0100
+++ slrn-0.9.8.1pl1/src/art.c	2006-03-07 20:32:04.375072500 +0100
@@ -5064,8 +5064,8 @@
 				       (unsigned char *) l->from + strlen (l->from),
 				       &st))
 #else
-		 && (NULL != SLsearch_forward (st, (SLuchar_Type *) l->subject,
-					       (SLuchar_Type *) l->subject + strlen (l->subject)))
+		 && (NULL != SLsearch_forward (st, (SLuchar_Type *) l->from,
+					       (SLuchar_Type *) l->from + strlen (l->from)))
 #endif
 		     )
 	  break;
