diff -Nur slrn-0.9.8.0.orig/src/misc.c slrn-0.9.8.0/src/misc.c
--- slrn-0.9.8.0.orig/src/misc.c	2003-08-25 17:00:16.000000000 +0200
+++ slrn-0.9.8.0/src/misc.c	2003-11-22 18:22:26.000000000 +0100
@@ -836,6 +836,17 @@
 #endif /* VMS */
 }
 
+void slrn_make_home_etc_filename(char *name, char *file, size_t n)
+{
+	char *home_etc = getenv("HOME_ETC");
+	
+	if (!home_etc) slrn_make_home_filename(name, file, n);
+	else {
+		*file = 0;
+		slrn_dircat(home_etc, name, file, n);
+	}
+}
+
 /*}}}*/
 
 void slrn_make_home_dirname (char *name, char *dir, size_t n) /*{{{*/
diff -Nur slrn-0.9.8.0.orig/src/misc.h slrn-0.9.8.0/src/misc.h
--- slrn-0.9.8.0.orig/src/misc.h	2003-11-22 18:11:46.000000000 +0100
+++ slrn-0.9.8.0/src/misc.h	2003-11-22 18:22:49.000000000 +0100
@@ -28,6 +28,7 @@
 #include "vfile.h"
 
 extern void slrn_make_home_filename (char *, char *, size_t);
+extern void slrn_make_home_etc_filename (char *, char *, size_t);
 extern void slrn_make_home_dirname (char *, char *, size_t);
 extern void slrn_redraw (void);
 extern void slrn_update_screen (void);
diff -Nur slrn-0.9.8.0.orig/src/slrn.c slrn-0.9.8.0/src/slrn.c
--- slrn-0.9.8.0.orig/src/slrn.c	2003-07-29 15:03:28.000000000 +0200
+++ slrn-0.9.8.0/src/slrn.c	2003-11-22 18:25:01.000000000 +0100
@@ -244,11 +244,11 @@
 #endif
    
 #if SLRN_HAS_RNLOCK
-   slrn_make_home_filename (".newsrc", file_rn, sizeof (file_rn));
+   slrn_make_home_etc_filename (".newsrc", file_rn, sizeof (file_rn));
    if (0 == strcmp (file_rn, Slrn_Newsrc_File))
      {
 	rnlock = 1;
-	slrn_make_home_filename (".rnlock", file_rn, sizeof (file_rn));
+	slrn_make_home_etc_filename (".rnlock", file_rn, sizeof (file_rn));
      }
 #endif
 
@@ -982,7 +982,7 @@
    if (Slrn_Score_File == NULL)
      return;
    
-   slrn_make_home_filename (Slrn_Score_File, file, sizeof (file));
+   slrn_make_home_etc_filename (Slrn_Score_File, file, sizeof (file));
    
    if (1 != slrn_file_exists (file))
      {
@@ -1184,7 +1184,7 @@
    
    if (init_file == NULL)
      {
-	slrn_make_home_filename (SLRN_USER_SLRNRC_FILENAME, file, sizeof (file));
+	slrn_make_home_etc_filename (SLRN_USER_SLRNRC_FILENAME, file, sizeof (file));
 	init_file = file;
      }
    
@@ -1312,7 +1312,7 @@
 #else
 	Slrn_Newsrc_File = ".jnewsrc";
 #endif
-	slrn_make_home_filename (Slrn_Newsrc_File, file, sizeof (file));
+	slrn_make_home_etc_filename (Slrn_Newsrc_File, file, sizeof (file));
 	Slrn_Newsrc_File = slrn_safe_strmalloc (file);
      }
    
