diff -Nur slrn-0.9.8.0.orig/macros/uname.sl slrn-0.9.8.0/macros/uname.sl
--- slrn-0.9.8.0.orig/macros/uname.sl	1970-01-01 01:00:00.000000000 +0100
+++ slrn-0.9.8.0/macros/uname.sl	2003-08-26 20:30:42.442283944 +0200
@@ -0,0 +1,4 @@
+% example of customizing User-Agent header
+
+set_string_variable("system_type", uname().sysname + " " + uname().release
+    + "/" + uname().machine);
--- slrn/macros/INSTFILES.orig	2007-11-10 06:07:20.000000000 +0100
+++ slrn/macros/INSTFILES	2008-07-25 20:24:46.864745949 +0200
@@ -13,3 +13,4 @@
 util.sl
 varset.sl
 xcomment.sl
+uname.sl
diff -Nur slrn-0.9.8.0.orig/src/misc.h slrn-0.9.8.0/src/misc.h
--- slrn-0.9.8.0.orig/src/misc.h	2002-10-06 00:19:51.000000000 +0200
+++ slrn-0.9.8.0/src/misc.h	2003-08-26 20:30:42.443283792 +0200
@@ -81,6 +81,8 @@
 #endif
    char *posting_host;		       /* FQDN or NULL */
    char *login_name;
+   char *user_agent;
+   char *system_type;
 }
 Slrn_User_Info_Type;
 
--- slrn/src/post.c.orig	2008-07-11 07:54:20.000000000 +0200
+++ slrn/src/post.c	2008-07-25 20:25:59.896750139 +0200
@@ -666,9 +666,16 @@
    int newsgroups_found=0, subject_found=0, followupto_found=0;
    Slrn_Mime_Error_Obj *err=NULL;
    Slrn_Mime_Error_Obj *mime_err;
-   char *tmp, *system_os_name;
+   char *tmp, *system_os_name, *User_Agent;
 
+   if (Slrn_User_Info.system_type != NULL)
+     system_os_name = Slrn_User_Info.system_type;
+   else
    system_os_name = slrn_get_os_name ();
+   if (Slrn_User_Info.user_agent != NULL)
+     User_Agent = Slrn_User_Info.user_agent;
+   else
+     User_Agent = "slrn";
 
 #if SLRN_HAS_STRICT_FROM
    if (NULL != (tmp = slrn_make_from_string ()))
@@ -825,7 +832,7 @@
 
    if ((NULL == (tmp = slrn_gen_date_header ()))
        || (NULL == slrn_append_to_header (a, tmp, 1))
-       || (NULL == (tmp = slrn_strdup_printf("User-Agent: slrn/%s (%s)", Slrn_Version_String, system_os_name)))
+       || (NULL == (tmp = slrn_strdup_printf("User-Agent: %s/%s (%s)", User_Agent, Slrn_Version_String, system_os_name)))
        || (NULL == slrn_append_to_header (a, tmp, 1))
        || (NULL == slrn_append_to_header (a, NULL,0)))   /* separator */
      {
--- slrn/src/startup.c.orig	2008-07-10 07:15:22.000000000 +0200
+++ slrn/src/startup.c	2008-07-25 20:26:23.960750978 +0200
@@ -756,6 +756,8 @@
      {"followup_custom_headers", &Slrn_Followup_Custom_Headers, NULL},
      {"reply_custom_headers", &Slrn_Reply_Custom_Headers, NULL},
      {"supersedes_custom_headers", &Slrn_Supersedes_Custom_Headers, NULL},
+     {"user_agent", &Slrn_User_Info.user_agent},
+     {"system_type", &Slrn_User_Info.system_type},
 #if SLRN_HAS_GROUPLENS
      {"grouplens_pseudoname", &Slrn_GroupLens_Pseudoname, NULL},
      {"grouplens_host", &Slrn_GroupLens_Host, NULL},
