diff -urNp -x '*.orig' slrn-1.0.3.org/src/art.c slrn-1.0.3/src/art.c
--- slrn-1.0.3.org/src/art.c	2016-10-24 00:34:16.000000000 +0200
+++ slrn-1.0.3/src/art.c	2021-10-03 22:48:58.844392038 +0200
@@ -7672,20 +7672,28 @@ void slrn_init_article_mode (void) /*{{{
 #else
    SLkm_define_key  ("\033\033[A", (FVOID_STAR) art_lineup, Slrn_Article_Keymap);
    SLkm_define_key  ("\033\033OA", (FVOID_STAR) art_lineup, Slrn_Article_Keymap);
+   SLkm_define_key  ("\033^(ku)", (FVOID_STAR) art_lineup, Slrn_Article_Keymap);
    SLkm_define_key  ("\033\033[B", (FVOID_STAR) art_linedn, Slrn_Article_Keymap);
    SLkm_define_key  ("\033\033OB", (FVOID_STAR) art_linedn, Slrn_Article_Keymap);
+   SLkm_define_key  ("\033^(kd)", (FVOID_STAR) art_linedn, Slrn_Article_Keymap);
    SLkm_define_key  ("\033\033[C", (FVOID_STAR) skip_to_next_group_1, Slrn_Article_Keymap);
    SLkm_define_key  ("\033\033OC", (FVOID_STAR) skip_to_next_group_1, Slrn_Article_Keymap);
+   SLkm_define_key  ("\033^(kr)", (FVOID_STAR) skip_to_next_group_1, Slrn_Article_Keymap);
    SLkm_define_key  ("\033\033[D", (FVOID_STAR) skip_to_prev_group, Slrn_Article_Keymap);
    SLkm_define_key  ("\033\033OD", (FVOID_STAR) skip_to_prev_group, Slrn_Article_Keymap);
+   SLkm_define_key  ("\033^(kl)", (FVOID_STAR) skip_to_prev_group, Slrn_Article_Keymap);
    SLkm_define_key  ("\033[A", (FVOID_STAR) header_up, Slrn_Article_Keymap);
    SLkm_define_key  ("\033OA", (FVOID_STAR) header_up, Slrn_Article_Keymap);
+   SLkm_define_key  ("^(ku)", (FVOID_STAR) header_up, Slrn_Article_Keymap);
    SLkm_define_key  ("\033[B", (FVOID_STAR) header_down, Slrn_Article_Keymap);
    SLkm_define_key  ("\033OB", (FVOID_STAR) header_down, Slrn_Article_Keymap);
+   SLkm_define_key  ("^(kd)", (FVOID_STAR) header_down, Slrn_Article_Keymap);
    SLkm_define_key  ("\033[C", (FVOID_STAR) art_right, Slrn_Article_Keymap);
    SLkm_define_key  ("\033OC", (FVOID_STAR) art_right, Slrn_Article_Keymap);
+   SLkm_define_key  ("^(kr)", (FVOID_STAR) art_right, Slrn_Article_Keymap);
    SLkm_define_key  ("\033[D", (FVOID_STAR) art_left, Slrn_Article_Keymap);
    SLkm_define_key  ("\033OD", (FVOID_STAR) art_left, Slrn_Article_Keymap);
+   SLkm_define_key  ("^(kl)", (FVOID_STAR) art_left, Slrn_Article_Keymap);
 #endif
    SLkm_define_key  ("^U", (FVOID_STAR) header_pageup, Slrn_Article_Keymap);
    SLkm_define_key  ("\033V", (FVOID_STAR) header_pageup, Slrn_Article_Keymap);
@@ -7695,10 +7703,8 @@ void slrn_init_article_mode (void) /*{{{
    SLkm_define_key  ("^@Q", (FVOID_STAR) header_pagedn, Slrn_Article_Keymap);
    SLkm_define_key  ("\xE0Q", (FVOID_STAR) header_pagedn, Slrn_Article_Keymap);
 #else
-   SLkm_define_key  ("\033[5~", (FVOID_STAR) header_pageup, Slrn_Article_Keymap);
-   SLkm_define_key  ("\033[I", (FVOID_STAR) header_pageup, Slrn_Article_Keymap);
-   SLkm_define_key  ("\033[6~", (FVOID_STAR) header_pagedn, Slrn_Article_Keymap);
-   SLkm_define_key  ("\033[G", (FVOID_STAR) header_pagedn, Slrn_Article_Keymap);
+   SLkm_define_key  ("^(kP)", (FVOID_STAR) header_pageup, Slrn_Article_Keymap);
+   SLkm_define_key  ("^(kN)", (FVOID_STAR) header_pagedn, Slrn_Article_Keymap);
 #endif
    SLkm_define_key  ("^D", (FVOID_STAR) header_pagedn, Slrn_Article_Keymap);
    SLkm_define_key  ("^V", (FVOID_STAR) header_pagedn, Slrn_Article_Keymap);
diff -urNp -x '*.orig' slrn-1.0.3.org/src/group.c slrn-1.0.3/src/group.c
--- slrn-1.0.3.org/src/group.c	2016-10-24 00:34:16.000000000 +0200
+++ slrn-1.0.3/src/group.c	2021-10-03 22:48:58.844392038 +0200
@@ -2027,8 +2027,10 @@ void slrn_init_group_mode (void) /*{{{*/
    SLkm_define_key ("\0330", (FVOID_STAR) slrn_digit_arg, Slrn_Group_Keymap);
    SLkm_define_key  ("^K\033[A", (FVOID_STAR) group_bob, Slrn_Group_Keymap);
    SLkm_define_key  ("^K\033OA", (FVOID_STAR) group_bob, Slrn_Group_Keymap);
+   SLkm_define_key  ("^K^(ku)", (FVOID_STAR) group_bob, Slrn_Group_Keymap);
    SLkm_define_key  ("^K\033[B", (FVOID_STAR) group_eob, Slrn_Group_Keymap);
    SLkm_define_key  ("^K\033OB", (FVOID_STAR) group_eob, Slrn_Group_Keymap);
+   SLkm_define_key  ("^K^(kd)", (FVOID_STAR) group_eob, Slrn_Group_Keymap);
    SLkm_define_key  ("\033a", (FVOID_STAR) toggle_group_formats, Slrn_Group_Keymap);
    SLkm_define_key  ("\033>", (FVOID_STAR) group_eob, Slrn_Group_Keymap);
    SLkm_define_key  ("\033<", (FVOID_STAR) group_bob, Slrn_Group_Keymap);
@@ -2040,10 +2042,8 @@ void slrn_init_group_mode (void) /*{{{*/
    SLkm_define_key  ("^@I", (FVOID_STAR) group_pageup, Slrn_Group_Keymap);
    SLkm_define_key  ("\xE0I", (FVOID_STAR) group_pageup, Slrn_Group_Keymap);
 #else
-   SLkm_define_key  ("\033[6~", (FVOID_STAR) group_pagedown, Slrn_Group_Keymap);
-   SLkm_define_key  ("\033[G", (FVOID_STAR) group_pagedown, Slrn_Group_Keymap);
-   SLkm_define_key  ("\033[5~", (FVOID_STAR) group_pageup, Slrn_Group_Keymap);
-   SLkm_define_key  ("\033[I", (FVOID_STAR) group_pageup, Slrn_Group_Keymap);
+   SLkm_define_key  ("^(kN)", (FVOID_STAR) group_pagedown, Slrn_Group_Keymap);
+   SLkm_define_key  ("^(kP)", (FVOID_STAR) group_pageup, Slrn_Group_Keymap);
 #endif
    SLkm_define_key  ("m", (FVOID_STAR) move_group_cmd, Slrn_Group_Keymap);
    SLkm_define_key  ("^U", (FVOID_STAR) group_pageup, Slrn_Group_Keymap);
@@ -2078,8 +2078,10 @@ void slrn_init_group_mode (void) /*{{{*/
 #else
    SLkm_define_key  ("\033[A", (FVOID_STAR) group_up, Slrn_Group_Keymap);
    SLkm_define_key  ("\033OA", (FVOID_STAR) group_up, Slrn_Group_Keymap);
+   SLkm_define_key  ("^(ku)", (FVOID_STAR) group_up, Slrn_Group_Keymap);
    SLkm_define_key  ("\033[B", (FVOID_STAR) group_down, Slrn_Group_Keymap);
    SLkm_define_key  ("\033OB", (FVOID_STAR) group_down, Slrn_Group_Keymap);
+   SLkm_define_key  ("^(kd)", (FVOID_STAR) group_down, Slrn_Group_Keymap);
 #endif
    SLkm_define_key  ("N", (FVOID_STAR) group_down, Slrn_Group_Keymap);
    SLkm_define_key  ("^N", (FVOID_STAR) group_down, Slrn_Group_Keymap);
diff -urNp -x '*.orig' slrn-1.0.3.org/src/menu.c slrn-1.0.3/src/menu.c
--- slrn-1.0.3.org/src/menu.c	2016-10-24 00:34:16.000000000 +0200
+++ slrn-1.0.3/src/menu.c	2021-10-03 22:48:58.844392038 +0200
@@ -783,16 +783,14 @@ static int init_select_list_mode (int se
    SLkm_define_key  ("^@Q", (FVOID_STAR) sl_pagedn, *kmap);
    SLkm_define_key  ("\xE0Q", (FVOID_STAR) sl_pagedn, *kmap);
 #else
-   SLkm_define_key  ("\033[5~", (FVOID_STAR) sl_pageup, *kmap);
-   SLkm_define_key  ("\033[6~", (FVOID_STAR) sl_pagedn, *kmap);
+   SLkm_define_key  ("^(kP)", (FVOID_STAR) sl_pageup, *kmap);
+   SLkm_define_key  ("^(kN)", (FVOID_STAR) sl_pagedn, *kmap);
    SLkm_define_key  ("\033[A", (FVOID_STAR) sl_up, *kmap);
    SLkm_define_key  ("\033OA", (FVOID_STAR) sl_up, *kmap);
+   SLkm_define_key  ("^(ku)", (FVOID_STAR) sl_up, *kmap);
    SLkm_define_key  ("\033[B", (FVOID_STAR) sl_down, *kmap);
    SLkm_define_key  ("\033OB", (FVOID_STAR) sl_down, *kmap);
-# ifdef __unix__
    SLkm_define_key  ("^(kd)", (FVOID_STAR) sl_down, *kmap);
-   SLkm_define_key  ("^(ku)", (FVOID_STAR) sl_up, *kmap);
-# endif
 #endif
 
    /* In select lists, these bindings collide with the new "jump" feature */
@@ -809,12 +807,10 @@ static int init_select_list_mode (int se
 #else
 	SLkm_define_key  ("\033[C", (FVOID_STAR) sl_right, Select_List_Keymap);
 	SLkm_define_key  ("\033OC", (FVOID_STAR) sl_right, Select_List_Keymap);
+	SLkm_define_key  ("^(kr)", (FVOID_STAR) sl_right, Select_List_Keymap);
 	SLkm_define_key  ("\033OD", (FVOID_STAR) sl_left, Select_List_Keymap);
 	SLkm_define_key  ("\033[D", (FVOID_STAR) sl_left, Select_List_Keymap);
-# ifdef __unix__
-	SLkm_define_key  ("^(kr)", (FVOID_STAR) sl_right, Select_List_Keymap);
 	SLkm_define_key  ("^(kl)", (FVOID_STAR) sl_left, Select_List_Keymap);
-# endif
 #endif
 	SLkm_define_key  ("\r", (FVOID_STAR) sl_select, Select_List_Keymap);
 	SLkm_define_key  ("^G", (FVOID_STAR) sl_cancel, Select_List_Keymap);
