diff -ur slrn.orig/autoconf/aclocal.m4 slrn/autoconf/aclocal.m4
--- slrn.orig/autoconf/aclocal.m4	2008-06-18 18:56:47.000000000 +0200
+++ slrn/autoconf/aclocal.m4	2008-08-01 14:55:08.000000000 +0200
@@ -850,9 +850,9 @@
 	 /usr/local/$1/include,/usr/local/$1/lib \
 	 /usr/local/include/$1,/usr/local/lib \
 	 /usr/local/include,/usr/local/lib \
-	 /usr/include/$1,/usr/lib \
-	 /usr/$1/include,/usr/$1/lib \
-	 /usr/include,/usr/lib \
+	 /usr/include/$1,/usr/lib64 \
+	 /usr/$1/include,/usr/$1/lib64 \
+	 /usr/include,/usr/lib64 \
 	 /opt/include/$1,/opt/lib \
 	 /opt/$1/include,/opt/$1/lib \
 	 /opt/include,/opt/lib"
