--- slrn-0.9.6.4pl4c.orig/src/slrn.c	Mon Mar 19 21:13:41 2001
+++ slrn-0.9.6.4pl4c/src/slrn.c	Fri Mar 30 17:14:28 2001
@@ -1120,6 +1120,6 @@
     * SLtt_get_terminfo call above fixed that.
     */
-   slrn_read_startup_file (file);      /* global file for all users */
+   slrn_read_startup_file ("/etc/slrn.rc");	/* global file for all users */
    slrn_read_startup_file (init_file);
 
    if (Slrn_Server_Id == 0) Slrn_Server_Id = Slrn_Default_Server_Obj;
