diff -ur slrn-0.9.7.3.orig/src/art.c slrn/src/art.c
--- slrn-0.9.7.3.orig/src/art.c	Sat Nov  3 21:48:57 2001
+++ slrn/src/art.c	Mon Jan  7 19:06:52 2002
@@ -3678,7 +3678,7 @@
        || (Slrn_Prefix_Arg_Ptr != NULL))
      {
 	Slrn_Prefix_Arg_Ptr = NULL;
-	if (slrn_read_input ("Subject: ", same_subject, NULL, 0, 0) <= 0) return;
+	if (slrn_read_input (_("Subject: "), same_subject, NULL, 0, 0) <= 0) return;
 	Same_Subject_Start_Header = Slrn_Current_Header;
      }
    SLsearch_init (same_subject, 1, 0, &st);
