diff -Nur slrn-0.9.8.0.orig/src/art.c slrn-0.9.8.0/src/art.c
--- slrn-0.9.8.0.orig/src/art.c	2003-08-25 16:59:16.000000000 +0200
+++ slrn-0.9.8.0/src/art.c	2003-08-26 20:21:55.098452392 +0200
@@ -7263,20 +7263,28 @@
 #else
    SLkm_define_key  ("\033\033[A", (FVOID_STAR) art_lineup, Slrn_Article_Keymap);
    SLkm_define_key  ("\033\033OA", (FVOID_STAR) art_lineup, Slrn_Article_Keymap);
+   SLkm_define_key  ("\033^(ku)", (FVOID_STAR) art_lineup, Slrn_Article_Keymap);
    SLkm_define_key  ("\033\033[B", (FVOID_STAR) art_linedn, Slrn_Article_Keymap);
    SLkm_define_key  ("\033\033OB", (FVOID_STAR) art_linedn, Slrn_Article_Keymap);
+   SLkm_define_key  ("\033^(kd)", (FVOID_STAR) art_linedn, Slrn_Article_Keymap);
    SLkm_define_key  ("\033\033[C", (FVOID_STAR) skip_to_next_group_1, Slrn_Article_Keymap);
    SLkm_define_key  ("\033\033OC", (FVOID_STAR) skip_to_next_group_1, Slrn_Article_Keymap);
+   SLkm_define_key  ("\033^(kr)", (FVOID_STAR) skip_to_next_group_1, Slrn_Article_Keymap);
    SLkm_define_key  ("\033\033[D", (FVOID_STAR) skip_to_prev_group, Slrn_Article_Keymap);
    SLkm_define_key  ("\033\033OD", (FVOID_STAR) skip_to_prev_group, Slrn_Article_Keymap);
+   SLkm_define_key  ("\033^(kl)", (FVOID_STAR) skip_to_prev_group, Slrn_Article_Keymap);
    SLkm_define_key  ("\033[A", (FVOID_STAR) header_up, Slrn_Article_Keymap);
    SLkm_define_key  ("\033OA", (FVOID_STAR) header_up, Slrn_Article_Keymap);
+   SLkm_define_key  ("^(ku)", (FVOID_STAR) header_up, Slrn_Article_Keymap);
    SLkm_define_key  ("\033[B", (FVOID_STAR) header_down, Slrn_Article_Keymap);
    SLkm_define_key  ("\033OB", (FVOID_STAR) header_down, Slrn_Article_Keymap);
+   SLkm_define_key  ("^(kd)", (FVOID_STAR) header_down, Slrn_Article_Keymap);
    SLkm_define_key  ("\033[C", (FVOID_STAR) art_right, Slrn_Article_Keymap);
    SLkm_define_key  ("\033OC", (FVOID_STAR) art_right, Slrn_Article_Keymap);
+   SLkm_define_key  ("^(kr)", (FVOID_STAR) art_right, Slrn_Article_Keymap);
    SLkm_define_key  ("\033[D", (FVOID_STAR) art_left, Slrn_Article_Keymap);
    SLkm_define_key  ("\033OD", (FVOID_STAR) art_left, Slrn_Article_Keymap);
+   SLkm_define_key  ("^(kl)", (FVOID_STAR) art_left, Slrn_Article_Keymap);
 #endif
    SLkm_define_key  ("^U", (FVOID_STAR) header_pageup, Slrn_Article_Keymap);
    SLkm_define_key  ("\033V", (FVOID_STAR) header_pageup, Slrn_Article_Keymap);
@@ -7286,10 +7294,8 @@
    SLkm_define_key  ("^@Q", (FVOID_STAR) header_pagedn, Slrn_Article_Keymap);
    SLkm_define_key  ("\xE0Q", (FVOID_STAR) header_pagedn, Slrn_Article_Keymap);
 #else
-   SLkm_define_key  ("\033[5~", (FVOID_STAR) header_pageup, Slrn_Article_Keymap);
-   SLkm_define_key  ("\033[I", (FVOID_STAR) header_pageup, Slrn_Article_Keymap);
-   SLkm_define_key  ("\033[6~", (FVOID_STAR) header_pagedn, Slrn_Article_Keymap);
-   SLkm_define_key  ("\033[G", (FVOID_STAR) header_pagedn, Slrn_Article_Keymap);
+   SLkm_define_key  ("^(kP)", (FVOID_STAR) header_pageup, Slrn_Article_Keymap);
+   SLkm_define_key  ("^(kN)", (FVOID_STAR) header_pagedn, Slrn_Article_Keymap);
 #endif
    SLkm_define_key  ("^D", (FVOID_STAR) header_pagedn, Slrn_Article_Keymap);
    SLkm_define_key  ("^V", (FVOID_STAR) header_pagedn, Slrn_Article_Keymap);
diff -Nur slrn-0.9.8.0.orig/src/group.c slrn-0.9.8.0/src/group.c
--- slrn-0.9.8.0.orig/src/group.c	2003-08-25 16:47:09.000000000 +0200
+++ slrn-0.9.8.0/src/group.c	2003-08-26 20:21:55.110450568 +0200
@@ -1992,8 +1992,10 @@
    SLkm_define_key ("\0330", (FVOID_STAR) slrn_digit_arg, Slrn_Group_Keymap);
    SLkm_define_key  ("^K\033[A", (FVOID_STAR) group_bob, Slrn_Group_Keymap);
    SLkm_define_key  ("^K\033OA", (FVOID_STAR) group_bob, Slrn_Group_Keymap);
+   SLkm_define_key  ("^K^(ku)", (FVOID_STAR) group_bob, Slrn_Group_Keymap);
    SLkm_define_key  ("^K\033[B", (FVOID_STAR) group_eob, Slrn_Group_Keymap);
    SLkm_define_key  ("^K\033OB", (FVOID_STAR) group_eob, Slrn_Group_Keymap);
+   SLkm_define_key  ("^K^(kd)", (FVOID_STAR) group_eob, Slrn_Group_Keymap);
    SLkm_define_key  ("\033a", (FVOID_STAR) toggle_group_formats, Slrn_Group_Keymap);
    SLkm_define_key  ("\033>", (FVOID_STAR) group_eob, Slrn_Group_Keymap);
    SLkm_define_key  ("\033<", (FVOID_STAR) group_bob, Slrn_Group_Keymap);
@@ -2005,10 +2007,8 @@
    SLkm_define_key  ("^@I", (FVOID_STAR) group_pageup, Slrn_Group_Keymap);
    SLkm_define_key  ("\xE0I", (FVOID_STAR) group_pageup, Slrn_Group_Keymap);   
 #else
-   SLkm_define_key  ("\033[6~", (FVOID_STAR) group_pagedown, Slrn_Group_Keymap);
-   SLkm_define_key  ("\033[G", (FVOID_STAR) group_pagedown, Slrn_Group_Keymap);
-   SLkm_define_key  ("\033[5~", (FVOID_STAR) group_pageup, Slrn_Group_Keymap);
-   SLkm_define_key  ("\033[I", (FVOID_STAR) group_pageup, Slrn_Group_Keymap);
+   SLkm_define_key  ("^(kN)", (FVOID_STAR) group_pagedown, Slrn_Group_Keymap);
+   SLkm_define_key  ("^(kP)", (FVOID_STAR) group_pageup, Slrn_Group_Keymap);
 #endif
    SLkm_define_key  ("m", (FVOID_STAR) move_group_cmd, Slrn_Group_Keymap);
    SLkm_define_key  ("^U", (FVOID_STAR) group_pageup, Slrn_Group_Keymap);
@@ -2043,8 +2043,10 @@
 #else
    SLkm_define_key  ("\033[A", (FVOID_STAR) group_up, Slrn_Group_Keymap);
    SLkm_define_key  ("\033OA", (FVOID_STAR) group_up, Slrn_Group_Keymap);
+   SLkm_define_key  ("^(ku)", (FVOID_STAR) group_up, Slrn_Group_Keymap);
    SLkm_define_key  ("\033[B", (FVOID_STAR) group_down, Slrn_Group_Keymap);
    SLkm_define_key  ("\033OB", (FVOID_STAR) group_down, Slrn_Group_Keymap);
+   SLkm_define_key  ("^(kd)", (FVOID_STAR) group_down, Slrn_Group_Keymap);
 #endif
    SLkm_define_key  ("N", (FVOID_STAR) group_down, Slrn_Group_Keymap);
    SLkm_define_key  ("^N", (FVOID_STAR) group_down, Slrn_Group_Keymap);
diff -Nur slrn-0.9.8.0.orig/src/menu.c slrn-0.9.8.0/src/menu.c
--- slrn-0.9.8.0.orig/src/menu.c	2002-06-26 18:57:34.000000000 +0200
+++ slrn-0.9.8.0/src/menu.c	2003-08-26 20:21:55.118449352 +0200
@@ -748,16 +748,14 @@
    SLkm_define_key  ("^@Q", (FVOID_STAR) sl_pagedn, *kmap);
    SLkm_define_key  ("\xE0Q", (FVOID_STAR) sl_pagedn, *kmap);
 #else
-   SLkm_define_key  ("\033[5~", (FVOID_STAR) sl_pageup, *kmap);
-   SLkm_define_key  ("\033[6~", (FVOID_STAR) sl_pagedn, *kmap);
+   SLkm_define_key  ("^(kP)", (FVOID_STAR) sl_pageup, *kmap);
+   SLkm_define_key  ("^(kN)", (FVOID_STAR) sl_pagedn, *kmap);
    SLkm_define_key  ("\033[A", (FVOID_STAR) sl_up, *kmap);
    SLkm_define_key  ("\033OA", (FVOID_STAR) sl_up, *kmap);
+   SLkm_define_key  ("^(ku)", (FVOID_STAR) sl_up, *kmap);
    SLkm_define_key  ("\033[B", (FVOID_STAR) sl_down, *kmap);
    SLkm_define_key  ("\033OB", (FVOID_STAR) sl_down, *kmap);
-# ifdef __unix__
    SLkm_define_key  ("^(kd)", (FVOID_STAR) sl_down, *kmap);
-   SLkm_define_key  ("^(ku)", (FVOID_STAR) sl_up, *kmap);
-# endif
 #endif
    
    /* In select lists, these bindings collide with the new "jump" feature */
@@ -774,12 +772,10 @@
 #else
 	SLkm_define_key  ("\033[C", (FVOID_STAR) sl_right, Select_List_Keymap);
 	SLkm_define_key  ("\033OC", (FVOID_STAR) sl_right, Select_List_Keymap);
+	SLkm_define_key  ("^(kr)", (FVOID_STAR) sl_right, Select_List_Keymap);
 	SLkm_define_key  ("\033OD", (FVOID_STAR) sl_left, Select_List_Keymap);
 	SLkm_define_key  ("\033[D", (FVOID_STAR) sl_left, Select_List_Keymap);
-# ifdef __unix__
-	SLkm_define_key  ("^(kr)", (FVOID_STAR) sl_right, Select_List_Keymap);
 	SLkm_define_key  ("^(kl)", (FVOID_STAR) sl_left, Select_List_Keymap);
-# endif
 #endif
 	SLkm_define_key  ("\r", (FVOID_STAR) sl_select, Select_List_Keymap);
 	SLkm_define_key  ("^G", (FVOID_STAR) sl_cancel, Select_List_Keymap);
