diff -ur slrn-0.9.7.3.orig/src/menu.c slrn-0.9.7.3/src/menu.c
--- slrn-0.9.7.3.orig/src/menu.c	Sat Nov  3 16:15:25 2001
+++ slrn-0.9.7.3/src/menu.c	Fri Jan 11 20:44:29 2002
@@ -133,7 +133,7 @@
    col = -1;
    while (m->menu_name != NULL)
      {
-	int dcol = 2 + strlen (m->menu_name);
+	int dcol = 2 + strlen (_(m->menu_name));
 	if ((want_col > col) 
 	    && (want_col <= col + dcol))
 	  break;
@@ -151,7 +151,7 @@
      {
 	slrn_set_color (color);
 	if (col) SLsmg_write_char (' ');
-	SLsmg_write_string (m->menu_name);
+	SLsmg_write_string (_(m->menu_name));
 	SLsmg_write_char (' ');
 	SLsmg_gotorc (0, col);
 	slrn_smg_refresh ();
diff -ur slrn-0.9.7.3.orig/src/post.c slrn-0.9.7.3/src/post.c
--- slrn-0.9.7.3.orig/src/post.c	Sat Nov  3 16:15:36 2001
+++ slrn-0.9.7.3/src/post.c	Fri Jan 11 20:44:25 2002
@@ -1209,7 +1209,10 @@
 	     linep[len - 1] = 0;
 #if SLRN_HAS_MIME
 	     if (Slrn_Use_Mime & MIME_DISPLAY)
-	       slrn_mime_header_encode (linep, sizeof (line) - 1);
+	       {
+		  slrn_mime_header_encode (linep, sizeof (line) - 1);
+		  len = strlen (linep);
+	       }
 #endif
 	  }
 
diff -ur slrn-0.9.7.3.orig/src/sltcp.c slrn-0.9.7.3/src/sltcp.c
--- slrn-0.9.7.3.orig/src/sltcp.c	Sat Nov  3 16:16:07 2001
+++ slrn-0.9.7.3/src/sltcp.c	Fri Jan 11 20:44:31 2002
@@ -730,14 +730,20 @@
 /*}}}*/
 
 #if !defined(VMS) && !defined(USE_WINSOCK_SLTCP)
-static int wait_for_input (int fd) /*{{{*/
+static int wait_for_input (SLTCP_Type *tcp) /*{{{*/
 {
    fd_set fds;
    struct timeval tv;
+   int fd = tcp->tcp_fd;
    
    if (fd == -1)
      return -1;
    
+#if SLTCP_HAS_SSL_SUPPORT
+   if (tcp->ssl != NULL && SSL_pending(tcp->ssl) > 0)
+     return 0;
+#endif
+   
    while (1)
      {
 	int ret;
@@ -814,7 +820,7 @@
     * slrn attempts to set signals to NOT restart system calls.  In such a 
     * case, the read below can be interrupted.
     */
-   if (-1 == wait_for_input (fd))
+   if (-1 == wait_for_input (tcp))
      return -1;
 #endif
 
