--- slrn-0.9.8.0/src/slrnfeat.h.orig	2003-07-30 15:36:32.000000000 +0200
+++ slrn-0.9.8.0/src/slrnfeat.h	2003-08-26 20:28:32.059105200 +0200
@@ -162,7 +162,7 @@
 /* The rest of the files apply to slrnpull. */
 
 /* This must be set to an absolute pathname. */
-#define SLRNPULL_ROOT_DIR	"/var/spool/news/slrnpull"
+#define SLRNPULL_ROOT_DIR	"/var/spool/slrnpull"
 
 /* The remaing variables are specified as relative names with respect to the
  * SLRNPULL_ROOT_DIR.
