diff -urNp -x '*.orig' slrn-1.0.3.org/macros/INSTFILES slrn-1.0.3/macros/INSTFILES
--- slrn-1.0.3.org/macros/INSTFILES	2016-10-24 00:34:16.000000000 +0200
+++ slrn-1.0.3/macros/INSTFILES	2021-10-03 22:48:59.187724986 +0200
@@ -15,3 +15,4 @@ varset.sl
 xcomment.sl
 mime.sl
 mailcap.sl
+uname.sl
diff -urNp -x '*.orig' slrn-1.0.3.org/macros/uname.sl slrn-1.0.3/macros/uname.sl
--- slrn-1.0.3.org/macros/uname.sl	1970-01-01 01:00:00.000000000 +0100
+++ slrn-1.0.3/macros/uname.sl	2021-10-03 22:48:59.187724986 +0200
@@ -0,0 +1,4 @@
+% example of customizing User-Agent header
+
+set_string_variable("system_type", uname().sysname + " " + uname().release
+    + "/" + uname().machine);
diff -urNp -x '*.orig' slrn-1.0.3.org/src/misc.h slrn-1.0.3/src/misc.h
--- slrn-1.0.3.org/src/misc.h	2016-10-24 00:34:16.000000000 +0200
+++ slrn-1.0.3/src/misc.h	2021-10-03 22:48:59.187724986 +0200
@@ -83,6 +83,8 @@ typedef struct
 #endif
    char *posting_host;		       /* FQDN or NULL */
    char *login_name;
+   char *user_agent;
+   char *system_type;
 }
 Slrn_User_Info_Type;
 
diff -urNp -x '*.orig' slrn-1.0.3.org/src/post.c slrn-1.0.3/src/post.c
--- slrn-1.0.3.org/src/post.c	2016-10-24 00:34:16.000000000 +0200
+++ slrn-1.0.3/src/post.c	2021-10-03 22:48:59.187724986 +0200
@@ -662,9 +662,16 @@ static int prepare_header (VFILE *vp, un
    int newsgroups_found=0, subject_found=0, followupto_found=0;
    Slrn_Mime_Error_Obj *err=NULL;
    Slrn_Mime_Error_Obj *mime_err;
-   char *tmp, *system_os_name;
+   char *tmp, *system_os_name, *User_Agent;
 
+   if (Slrn_User_Info.system_type != NULL)
+     system_os_name = Slrn_User_Info.system_type;
+   else
    system_os_name = slrn_get_os_name ();
+   if (Slrn_User_Info.user_agent != NULL)
+     User_Agent = Slrn_User_Info.user_agent;
+   else
+     User_Agent = "slrn";
 
 #if SLRN_HAS_STRICT_FROM
    if (NULL != (tmp = slrn_make_from_header ()))
@@ -821,7 +828,7 @@ static int prepare_header (VFILE *vp, un
 
    if ((NULL == (tmp = slrn_gen_date_header ()))
        || (NULL == slrn_append_to_header (a, tmp, 1))
-       || (NULL == (tmp = slrn_strdup_printf("User-Agent: slrn/%s (%s)", Slrn_Version_String, system_os_name)))
+       || (NULL == (tmp = slrn_strdup_printf("User-Agent: %s/%s (%s)", User_Agent, Slrn_Version_String, system_os_name)))
        || (NULL == slrn_append_to_header (a, tmp, 1))
        || (NULL == slrn_append_to_header (a, NULL,0)))   /* separator */
      {
diff -urNp -x '*.orig' slrn-1.0.3.org/src/startup.c slrn-1.0.3/src/startup.c
--- slrn-1.0.3.org/src/startup.c	2016-10-24 00:34:16.000000000 +0200
+++ slrn-1.0.3/src/startup.c	2021-10-03 22:48:59.191058315 +0200
@@ -766,6 +766,8 @@ Slrn_Str_Var_Type Slrn_Str_Variables []
      {"followup_custom_headers", &Slrn_Followup_Custom_Headers, NULL},
      {"reply_custom_headers", &Slrn_Reply_Custom_Headers, NULL},
      {"supersedes_custom_headers", &Slrn_Supersedes_Custom_Headers, NULL},
+     {"user_agent", &Slrn_User_Info.user_agent},
+     {"system_type", &Slrn_User_Info.system_type},
 #if SLRN_HAS_GROUPLENS
      {"grouplens_pseudoname", &Slrn_GroupLens_Pseudoname, NULL},
      {"grouplens_host", &Slrn_GroupLens_Host, NULL},
